common/persistence/versionHistory.go

```diff
@@ -458,6 +458,11 @@ func (h *VersionHistories) AddVersionHistory(
 	if newLastItem.version > currentLastItem.version {
 		currentBranchChanged = true
 		h.currentBranchIndex = newVersionHistoryIndex
+	} else if newLastItem.version == currentLastItem.version {
+		if newLastItem.eventID > currentLastItem.eventID {
+			currentBranchChanged = true
```

Comment:
why in this case the current branch changed?

Reply:
it has longer history